src/Symfony/Component/Messenger/Tests/Command/FailedMessagesRetryCommandTest.php

```diff
@@ -46,4 +50,52 @@ public function testBasicRun()
 
         $this->assertStringContainsString('[OK]', $tester->getDisplay());
     }
+
+    public function testExceptionOnRetry()
+    {
+        $receiver = $this->createMock(ListableReceiverInterface::class);
+        $receiver->expects($this->once())->method('find')->with(10)->willReturn(new Envelope(new \stdClass()));
+        // message will eventually be ack'ed in Worker
+        $receiver->expects($this->once())->method('ack');
+
+        $dispatcher = $this->createMock(EventDispatcherInterface::class);
+        $bus = $this->createMock(MessageBusInterface::class);
+        // the bus should be called in the worker
+        $bus->expects($this->at(0))
+            ->method('dispatch')
+            ->with($this->callback(function (Envelope $envelope) {
+                $lastReceivedStamp = $envelope->last(ReceivedStamp::class);
+
+                return $lastReceivedStamp instanceof ReceivedStamp &&
+                    \is_string($lastReceivedStamp->getTransportName());
+            }))
+            ->will($this->throwException(new Exception('Mock test exception')));
+
+        $bus->expects($this->at(1))
+            ->method('dispatch')
+            ->with($this->callback(function (Envelope $envelope) {
+                $lastRedeliveryStamp = $envelope->last(RedeliveryStamp::class);
+
+                return $lastRedeliveryStamp instanceof RedeliveryStamp &&
+                    \is_string($lastRedeliveryStamp->getExceptionMessage()) &&
+                    $lastRedeliveryStamp->getFlattenException() instanceof FlattenException;
```

Comment:
@fabpot Do you want me to update this to be one line too?

Reply:
This one looks good to me.